src/content/mod_configuration.cpp

```diff
@@ -221,10 +224,19 @@ void ModConfiguration::checkConflictsAndDeps()
 
 void ModConfiguration::resolveDependencies()
 {
-	// Step 1: Compile a list of the mod names we're working with
+	// Compile a list of the mod names we're working with
 	std::set<std::string> modnames;
-	for (const ModSpec &mod : m_unsatisfied_mods) {
-		modnames.insert(mod.name);
+	std::optional<ModSpec> first_mod_spec, last_mod_spec;
+	for (ModSpec &mod : m_unsatisfied_mods) {
+		if (mod.name == m_first_mod) {
+			first_mod_spec = mod;
+		} else if (mod.name == m_last_mod) {
+			// only non optional depends have to be check for last mod
+			mod.unsatisfied_depends = mod.depends;
+			last_mod_spec = mod;
+		} else {
+			modnames.insert(mod.name);
+		}
```

Comment:
what's the significance of removing first and last mod from `modnames`?

AFAICT this means nobody can depend on them, which makes sense for last_mod. but not necessarily for first_mod

Reply:
This is used just for optional depends, so it's fine to not add it for the first mod - if any mods optionally depend on it, these optional depends are automatically fulfilled.